cmd/vcert/commands.go

```diff
@@ -1154,6 +1163,73 @@ func doCommandRenew1(c *cli.Context) error {
 	return nil
 }
 
+func doCommandSshGetConfig(c *cli.Context) error {
+
+	err := validateGetSshConfigFlags(c.Command.Name)
+
+	if err != nil {
+		return err
+	}
+
+	err = setTLSConfig()
+	if err != nil {
+		return err
+	}
+
+	cfg, err := buildConfig(c, &flags)
+	if err != nil {
+		return fmt.Errorf("failed to build vcert config: %s", err)
+	}
+
+	connector, err := vcert.NewClient(&cfg)
+
+	if err != nil {
+		strErr := (err).Error()
+		if strErr != "vcert error: your data contains problems: auth error: failed to authenticate: can't determine valid credentials set" {
+			logf("Unable to build connector for %s: %s", cfg.ConnectorType, err)
+		} else {
+			logf("Successfully built connector for %s", cfg.ConnectorType)
+		}
+	} else {
+		logf("Successfully built connector for %s", cfg.ConnectorType)
+	}
+
+	err = connector.Ping()
+
+	if err != nil {
+		logf("Unable to connect to %s: %s", cfg.ConnectorType, err)
+	} else {
+		logf("Successfully connected to %s", cfg.ConnectorType)
+	}
+
+	req := &certificate.SshCaTemplateRequest{}
+	if flags.sshCertTemplate != "" {
+		req.DN = flags.sshCertTemplate
+	}
+	if flags.sshCertGuid != "" {
+		req.Guid = flags.sshCertGuid
+	}
+
+	conf, err := connector.RetrieveSshConfig(req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println()
+	fmt.Println("CA public key:")
+	fmt.Println(conf.CaPublicKey)
```

Comment:
It is good that we print the CA public key. Do you think it is possible to also store it as a file? This will be the most performed action by the users, as OpenSSH needs it in a file. So, if users specifie output file, then store it (0600), otherwise just print it.

Reply:
fixed on the most recent commits